### ISSUE TITLE
code-server module not loading settings

### ISSUE THREAD

**Issue by Conor-Behard333** [coder/modules#118]
Hi,
I am using the code server module and the settings are not being ported over to the code-server user space
Here is the tf module for my template: 
```
module "code-server" { 
  source = "https://registry.coder.com/modules/code-server" 
  agent_id = coder_agent.main.id 
  folder = "/home/coder/workspace/repo" 
  settings = { "workbench.colorTheme" = "Default Light Modern", "workbench.iconTheme" = "vscode-icons", "files.autoSave" = "onFocusChange", "editor.rulers" = "[80,120]", "update.mode" = "none", } 
  extensions = [ "ms-azuretools.vscode-docker", "sugatoray.vscode-git-extension-pack", "golang.Go", "ms-python.python", "redhat.vscode-yaml", "timonwong.shellcheck", "garlicbreadcleric.pandoc-markdown-syntax", "zaaack.markdown-editor", "hashicorp.terraform", "ms-kubernetes-tools.vscode-kubernetes-tools" ]
 }
```

After a little investigating into this script: https://github.com/coder/modules/blob/main/code-server/run.sh
Looking at the logs of my workspace "⚙️ Creating settings file..." is never outputted which means 
```
if [ ! -f ~/.local/share/code-server/Machine/settings.json ]; then
```
 is returning true. 

Not sure why this is as when I destroy the workspace it should be gone and recreating it still causes this issue.

**Comment by matifali** [comment]
@Conor-Behard333 can you check if there is a `~/.local/share/code-server/Machine/settings.json` file by running,

```shell
cat ~/.local/share/code-server/Machine/settings.json
```

**Comment by Conor-Behard333** [comment]
when I change a setting through vscode the settings get added to  ~/.local/share/code-server/User/settings.json

**Comment by matifali** [comment]
@Conor-Behard333 also on a fresh workspace do you have both ` ~/.local/share/code-server/User/settings.json` and `~/.local/share/code-server/Machine/settings.json`?

I am thinking if one of them is empty, we may probably add a new option to the module with the name `settings_scope` that defaults to `Machine` to keep the original behavior but allows passing `User` to set Settings as the User.

cc: @code-asher if you can provide more context here on what is happening.

**Comment by Conor-Behard333** [comment]
on a fresh workspace I have ~/.local/share/code-server/Machine/settings.json but ~/.local/share/code-server/User/settings.json is only created once I make a change in vscode (change the theme for example)

**Comment by code-asher** [comment]
A few observations:

- When you make manual changes from inside code-server, it will write to `User` by default; if you change the tab to `Remote` then it will write to `Machine` instead.
- Not all settings will work with `Machine`.  For example `update.mode` seems to only be valid for `User` settings and placing it in `Machine` will have no effect.  So a `scope` option will probably be necessary (or it will need to default to `User`).
- [Coder writes](https://github.com/coder/coder/blob/5b122d108e101219e21a05229c5959108b359e78/cli/gitauth/vscode.go#L21-L28) `Machine` settings which is why you already have it on a fresh workspace.  Maybe the module should try to merge settings instead of skipping if the file already exists?  Perhaps we can use `Node` or `jq` to do this.

**Comment by Conor-Behard333** [comment]
Any updates on this?

**Comment by matifali** [comment]
Closed in #123
